staff/asier/react-spotify-api-promises/scripts/main.jsx

```diff
@@ -0,0 +1,312 @@
+'use strict'
+
+// https://stackoverflow.com/questions/24502898/show-or-hide-element
+
+
+class SpotyApp extends React.Component {
+    constructor() {
+        super()
+        this.state = {
+            artists: [],
+            showArtists: true,
+            albums: [],
+            showAlbums: false,
+            songs: [],
+            showSongs: false,
+            songId: []
+        }

```

Comment:
MAY just the arrays length work as a flag to show / hide their corresponding panels in the view? that is, reseting the arrays when a new search is done (which also frees caching) and using later on the length as flags:

// in constructor
this.state = {
artists: [],
albums: [],
songs: [],
songId: []
}

// when deciding to show - or not - a panel in render

this.state.artists.length > 0? // then show panel : // otherwise not
// same for other panels...

// when doing a new search
this.setState({
artists: [],
albums: [],
songs: [],
songId: []
})

Reply:
@manuelbarzi  I see what you mean. But I think it might cause a problem with the "go back to artists" button... For now, I'm committing the other changes and tomorrow we can review this and the two API issues that I don't fully understand